chsdi/views/catalog.py

```diff
@@ -87,8 +87,6 @@ def cleanNode(node):
                     path = node['path'].split('/')
                     if len(path) > 0:
                         root = path[0]
-                else:
-                    return nodes_final
```

Comment:
I share the same concern that Gilbert. I need proof.

The best way to test that is to extract the code that creates the trees -> functional tests given a list of entries and nodes.
We could probaly rewrite this piece of code as well and use a recursive function.